src/cli/cli.cpp

```diff
@@ -287,29 +287,42 @@ template <> otError Interpreter::Process<Cmd("reset")>(Arg aArgs[])
 
 void Interpreter::ProcessLine(char *aBuf)
 {
-    Arg     args[kMaxArgs + 1];
-    otError error = OT_ERROR_NONE;
+    static const char *const kCmdFactoryReset = "factoryreset";
+    Arg                      args[kMaxArgs + 1];
+    otError                  error = OT_ERROR_NONE;
 
     OT_ASSERT(aBuf != nullptr);
 
+    args[0].Clear();
+
     if (!mInternalDebugCommand)
     {
         // Ignore the command if another command is pending.
-        VerifyOrExit(!mCommandIsPending, args[0].Clear());
-        mCommandIsPending = true;
-
-        VerifyOrExit(StringLength(aBuf, kMaxLineLength) <= kMaxLineLength - 1, error = OT_ERROR_PARSE);
+        if (!mCommandIsPending)
+        {
+            mCommandIsPending = true;
+            VerifyOrExit(StringLength(aBuf, kMaxLineLength) <= kMaxLineLength - 1, error = OT_ERROR_PARSE);
+        }
+        else
+        {
+            VerifyOrExit(StringLength(aBuf, kMaxLineLength) <= kMaxLineLength - 1, args[0].Clear());
```

Comment:
I guess we may not need `args[0].Clear()` here in the `StringLength()` check, since we do this already at the top?

But I think we do need it for next line where we do `ParseCmd()` (since `args` may get changed?) ... :)